Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,358 @@
+                        comparisonNode.Left = new ConstantNode()
+                        {
+                            Value = binaryLeftConstantExpression.Value,
+                            Type = GetKind(binaryLeftConstantExpression.Value.GetType())
+                        };
+                    }
+                }
+
+                if (binaryExpression.Right is MemberExpression binaryRightMemberExpression)
+                {
+                    if (binaryRightMemberExpression.Expression != null && binaryRightMemberExpression.Expression.NodeType == ExpressionType.Parameter)
+                    {
+                        comparisonNode.Right = new PropertyNode()
+                        {
+                            Name = GetColumnName(binaryRightMemberExpression, binaryRightMemberExpression.NodeType),
+                            Type = GetKind(binaryRightMemberExpression.Type)
+                        };
+                    }
+                    else
+                    {
+                        throw new NotSupportedException(binaryRightMemberExpression + " is null or not a supported node type.");
+                    }
+                }
+
+                if (binaryExpression.Right is ConstantExpression binaryRightConstantExpression)
+                {
+                    if (binaryRightConstantExpression.Value == null)
+                    {
+                        comparisonNode.Right = new ConstantNode()
+                        {
+                            Value = binaryRightConstantExpression.Value,
+                            Type = "string"
+                        };
+                    }
+                    else
+                    {
+                        comparisonNode.Right = new ConstantNode()
+                        {
+                            Value = binaryRightConstantExpression.Value,
+                            Type = GetKind(binaryRightConstantExpression.Value.GetType())
+                        };
+                    }
+                }
+
+                returnNode = comparisonNode;
+            }
+
+            return RealmLinqExpression.Create(returnNode);
+        }
+
+        protected override Expression VisitUnary(UnaryExpression node)
+        {
+            NegationNode returnNode = new NegationNode();
+            switch (node.NodeType)
+            {
+                case ExpressionType.Not:
+                    returnNode.Expression = Extract(node.Operand);
+                    break;
+                default:
+                    throw new NotSupportedException($"The unary operator '{node.NodeType}' is not supported");
+            }
+
+            return RealmLinqExpression.Create(returnNode);
+        }
+
+        protected override Expression VisitMember(MemberExpression node)
+        {
+            if (node.Expression != null && node.Expression.NodeType == ExpressionType.Parameter)
+            {
+                ComparisonNode comparisonNode = new EqualityNode();
```

Comment:
When is this used?

Reply:
boolean statements with no equality sign (e.g .Where(p => !p.IsInteresting))